featuretools/primitives/standard/transform_primitive.py

```diff
@@ -242,32 +243,81 @@ class Diff(TransformPrimitive):
     """Compute the difference between the value in a list and the
     previous value in that list.
 
+    Args:
+        periods (int): The number of periods by which to shift the index row.
+            Default is 0. Periods correspond to rows.
+
     Description:
         Given a list of values, compute the difference from the previous
         item in the list. The result for the first element of the list will
-        always be `NaN`. If the values are datetimes, the output will be a
-        timedelta.
+        always be `NaN`.
 
     Examples:
         >>> diff = Diff()
         >>> values = [1, 10, 3, 4, 15]
         >>> diff(values).tolist()
         [nan, 9.0, -7.0, 1.0, 11.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> values = [1, 2, 4, 7, 11, 16]
+        >>> diff_periods = Diff(periods = 1)
+        >>> diff_periods(values).tolist()
+        [nan, nan, 1.0, 2.0, 3.0, 4.0]
     """
 
     name = "diff"
-    input_types = [ColumnSchema(semantic_tags={"numeric"})]
+    input_types = [
+        [ColumnSchema(semantic_tags={"numeric"})],
+        [ColumnSchema(semantic_tags={"time_index"})],
```

Comment:
You'll still need to remove this column schema, or dfs will apply `Diff` to a datetime time index as well. If you're trying to specify that diff should only apply to numeric time indices, you could do `ColumnSchema(semantic_tags={"numeric", "time_index"})`

Reply:
forgot to remove this. Thanks!